enhancements/network/no_overlay_mode.md

```diff
@@ -0,0 +1,965 @@
+  - `spec.additionalRoutingCapabilities.providers`: `FRR`
+  - `spec.defaultNetwork.ovnKubernetesConfig.routeAdvertisements`: `Enabled`
+
+This is to be done at installation time for the default network and can be postponed to day 2 for CUDNs.
+At installation time this manifest should be provided in the `manifests` folder for the installer to apply on day 0:
+```yaml
+apiVersion: operator.openshift.io/v1
+kind: Network
+metadata:
+  name: cluster
+spec:
+  additionalRoutingCapabilities:
+    providers:
+    - FRR
+  defaultNetwork:
+    ovnKubernetesConfig:
+      routeAdvertisements: Enabled
+    type: OVNKubernetes
+```
+
+On day 2 we can patch `network.operator cluster` and CNO will take care of applying this change:
+```sh
+$ oc patch network.operator cluster --type merge --patch \
+'{
+   "spec":{
+      "additionalRoutingCapabilities":{
+         "providers":[
+            "FRR"
+         ]
+      },
+      "defaultNetwork":{
+         "ovnKubernetesConfig":{
+            "routeAdvertisements":"Enabled"
+         }
+      }
+   }
+}
+
+```
+#### No-overlay for the default network with unmanaged routing
+
+On day 0, in `operator.openshift.io/v1 Network` the cluster administrator should enable no-overlay mode for the default network and configure it with the necessary per-network parameters:
+  - `outboundSNAT`: `Enabled`,`Disabled`
+  - `routing`: `Unmanaged`
+
+In the prerequisite step, the cluster administrator already created a manifest for `operator.openshift.io/v1 Network` to enable BGP and placed it in the `manifests` folder. This manifest should therefore be extended to include the aforementioned parameters for the default network. For instance:
+```yaml
+apiVersion: operator.openshift.io/v1
+kind: Network
+metadata:
+  name: cluster
+spec:
+  additionalRoutingCapabilities:
+    providers:
+    - FRR
+  defaultNetwork:
+    ovnKubernetesConfig:
+      routeAdvertisements: Enabled
+      defaultNetworkTransport: NoOverlay
+      defaultNetworkNoOverlayOptions:
+        outboundSNAT: Disabled
+        routing: Unmanaged
+    type: OVNKubernetes
+```
+
+The cluster administrator should also provide in the `manifests` folder the necessary manifests for the `Unmanaged` scenario:
+  - FRRConfiguration CR (defines the BGP peering with the external router)
+  - RouteAdvertisements CR (defines which routes to advertise)
+
+##### Example: FRRConfiguration CR
+
+```yaml
+apiVersion: frrk8s.metallb.io/v1beta1
+kind: FRRConfiguration
+metadata:
+  name: external-bgp
+  namespace: openshift-frr-k8s
+  labels:
+    network: default
+spec:
+  bgp:
+    routers:
+    - asn: 64512
+      neighbors:
+      - address: 192.168.111.1  # External BGP router address
+        asn: 64512
+        disableMP: true
+        toReceive:
+          allowed:
+            mode: filtered
+            prefixes:
+            - prefix: 10.128.0.0/14  # Cluster pod subnet
+              ge: 23  # Must match the cluster's hostPrefix
```

Comment:
Ideally, we shouldn't need to add the prefixes up front. When the RA is processed OVNK will generate a FRRConfiguration that will include the prefixes. It there is a reason this is not viable, I would mention it explicitly.

Reply:
Same for the other examples.

Reply:
It's aligned with the u/s enhancement https://github.com/ovn-kubernetes/ovn-kubernetes/blob/master/docs/okeps/okep-5259-no-overlay.md#enable-no-overlay-mode-for-the-default-network-with-unmanaged-routing. 

It's viable to let the RA controller do the work. But it would be better to let users define it explicitly so they acknowledge that the routes will be imported.

Reply:
Because it is an implementation detail that we might change in the future and is also not really equivalent to what the RA does.

Reply:
@jcaamano so for cluster nodes to install BGP routes, the FRRConfiguration for the external BGP router would have `toReceive.allowed.mode: all` in the spec section:

```
spec:
  bgp:
    routers:
    - asn: 64512
      neighbors:
      - address: 192.168.111.1
        asn: 64512
        disableMP: true
        dualStackAddressFamily: false
        toReceive:
          allowed:
            mode: all
```
Is this how you saw it?

Reply:
> This is because the frr configurations generated by OVNK and then by frr-k8s do not fill in the cluster pod subnets:

But thats because it needs to be implemented as part no-overlay. They will be there once that happens.

Reply:
@pliurh I think we need an extra card in jira for this :)

Reply:
We actually had it before but we had to remove it:
https://github.com/ovn-kubernetes/ovn-kubernetes/commit/410550fbb05be947ca8b5ad44dc73c3d2141d2ac

```
... However in the future when we support
overlay-mode with BGP, we will need to re-add these routes ...
```

Several places of the upstream enhancement mention how these routes need to be imported, but don't explicitly mention how that happens or I guess it is proposing the user do this through example? From my point of view, it was clearly assumed it would be the RA controller doing this and I think that is how it should be. I guess that's something else we missed.

Reply:
/cc @tssurya 

I don't know, maybe it is just me being completely out of the loop here.

Reply:
Ok, we will let RA controller do this job.